internal/mapper/attrmapper/data_source_attributes.go

```diff
@@ -58,6 +60,46 @@ func (targetSlice DataSourceAttributes) Merge(mergeSlices ...DataSourceAttribute
 	return targetSlice, errResult
 }
 
+func (targetSlice DataSourceAttributes) MergeAttribute(path []string, attribute DataSourceAttribute, intermediateComputability schema.ComputedOptionalRequired) (DataSourceAttributes, error) {
+	var errResult error
+	if len(path) == 0 {
+		return targetSlice, errResult
+	}
+	for i, target := range targetSlice {
+		if target.GetName() == path[0] {
+
+			if len(path) > 1 {
+				nestedTarget, ok := target.(DataSourceNestedAttribute)
+				if !ok {
+					// TODO: error? there is a nested override for an attribute that is not a nested type
+					break
+				}
+
+				// The attribute we need to override is deeper nested, move up
+				nextPath := path[1:]
+
+				overriddenTarget, err := nestedTarget.NestedMerge(nextPath, attribute, intermediateComputability)
+				errResult = errors.Join(errResult, err)
+
+				targetSlice[i] = overriddenTarget
+
+			} else {
+				// No more path to traverse, apply merge, bidirectional
+				overriddenTarget, err := attribute.Merge(target)
+				errResult = errors.Join(errResult, err)
+				overriddenTarget, err = overriddenTarget.Merge(attribute)
+				errResult = errors.Join(errResult, err)
+
+				targetSlice[i] = overriddenTarget
+			}
+
+			break
+		}
+	}
+
+	return targetSlice, errResult
+}
```

Comment:
This logic will also need to take into account the scenario of a new parameter being merged (not aliased), similar to the original `Merge` function.

#### Example:

```go
isNewAttribute := true
for i, target := range targetSlice {
	if target.GetName() == path[0] {
		if len(path) > 1 {
			// Recurse logic
		} else {
			// Merge logic
		}

		isNewAttribute = false
		break
	}
}

if isNewAttribute {
	// Add this back to the original slice to avoid adding duplicate attributes from different mergeSlices
	targetSlice = append(targetSlice, attribute)
}
```